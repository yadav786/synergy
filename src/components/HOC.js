import React from "react";
export default function HOC(OldComponent) {
  // eslint-disable-next-line react/prefer-stateless-function
  return class extends React.Component {
    // For now, authorization is removed
    render() {
      return <OldComponent />;
    }
  };
}
