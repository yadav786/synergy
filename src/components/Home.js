import React, { useState, useEffect, useRef, useReducer } from 'react'
import { connect } from 'react-redux'
import 'bootstrap/dist/css/bootstrap.min.css'
import '../assets/css/App.css'

import { Card, CardBody, CardHeader, CardSubtitle, Spinner, Button } from 'reactstrap'
const Home = () => {

  const [initialize, onChangeState] = useState(0)

  const inputRef = useRef(null)
  const [data, getData] = useState(undefined)
  useEffect(() => {

    async function fetchData() {
      let result = await fetch('http://dummy.restapiexample.com/api/v1/employee/1')
      let allEmpData = await result.json()
      getData(allEmpData)
    }
    fetchData()
    console.log('onClick Button event inn useEffect', initialize)
  }, [])


  const initialState = { count: 1 }

  function reducer(state, action) {
    if (action.type === 'increment'){
      return { count: state.count + 1 }
    } else if (action.type === 'decrement'){
      return { count: state.count - 1 }
    }
    return state


  }

  const [store, dispatch] = useReducer(reducer, initialState)
  // console.log(dispatch);

  function onFocusChangeInput(){
    inputRef.current.focus()
  }

  return (
    <div className='container'>
      <Button onClick={(() => {
        dispatch({ type: 'increment' })
      })} color='info' outline>Dispatch Actions</Button>
      <input type='text' ref={inputRef} />
      <Button onClick={(() => {
        onFocusChangeInput()
      })}>Focus the Input</Button>
      <Button onClick={((initialize) => onChangeState(initialize + 1))} color='success'>INCREMENT</Button>
      {' '}
      <Button onClick={((initialize) => onChangeState(initialize - 1))} color='danger'>DECREMENT</Button>
      {' '}
      {(data === undefined) ? <Spinner style={{ width: '50%', height: '50%' }} color='primary' />
        :
        (
          <p>
        After onChangeState Called...
            { data.employee_name }
          </p>
        )
      }
      {/*
      <pre>
        { JSON.stringify(initialize) }
      </pre>
      */}
      <Card>
        <CardHeader>Query - Displaying all data</CardHeader>
        <CardBody>
         This is the Card Body
        </CardBody>
      </Card>
      <Card>
        <CardHeader>Query - Displaying data with args</CardHeader>
        <CardBody>
          <CardSubtitle>Viewing a user by id</CardSubtitle>

        </CardBody>
      </Card>
    </div>
  )
}

const mapStateToProps = (state) => {
  return state
}

export default connect(mapStateToProps, null)(Home)

