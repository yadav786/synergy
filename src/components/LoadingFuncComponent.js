import React from 'react';
 
export default function LoadingFuncComponent(){
  
    return(
			<p>LoadingFuncComponent</p>
	      )   
 
}  