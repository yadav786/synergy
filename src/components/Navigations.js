import React from 'react' 
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
const Navigations = (props) => {
  const { ...newProps } = props
  let isUserLoggedIn = newProps.fakeLoginData.isUserLoggedIn
  console.log('Navigations isUserLoggedIn', isUserLoggedIn)
  return (<ul>    
    { (!isUserLoggedIn) ?
      (
        <li><NavLink to='/'>Log In</NavLink></li>
      )
      :
      (
        <>
          <li><NavLink to='/home'>Home</NavLink></li>
          <li><NavLink to='/about'>About</NavLink></li>
          <li><NavLink to='/profile'>Profile</NavLink></li>
          <li><NavLink to='/contact'>Contact Us</NavLink></li>
          <li><NavLink to='/users'>User</NavLink></li>
        </ >   
      )
    }
  </ul>)

}

const mapStateToProps = (state) => {
  return state
}

export default connect(mapStateToProps)(Navigations)
