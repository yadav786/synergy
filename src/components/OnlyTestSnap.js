import React, { Component } from 'react'
export const OnlyTestSnap = (props) => {
  // console.log('props', props)
  const { items = [] } = props
  // console.log(items)
  if (!items.length){

    return (
     	<div>
     	 <p>No Items Found</p>
     	</div>
    )
  }

  if (items.length){
    return (
      <div>
        {items.map((value) =>
          (<p key={value}>
        	Items Found {value}
          </p>)
        )}

      </div>
    )
  }

}
