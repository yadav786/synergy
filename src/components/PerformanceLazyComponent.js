import React from 'react';
 
export default function PerformanceLazyComponent(){
    
    return( 
			<> 
			<p>PerformanceLazyComponent</p>
			</>
	      )   

}