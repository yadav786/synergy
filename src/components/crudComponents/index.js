import AddUser from "./AddUser";
import EditUser from "./EditUser";
import UserList from "./UserList";
import Users from "./Users";

export { AddUser, EditUser, UserList, Users };
