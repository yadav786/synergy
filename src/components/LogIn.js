import React, { Component } from 'react'
import { Card, CardBody, CardTitle, Button, FormGroup, Form, Input, Label } from 'reactstrap'
import { connect } from 'react-redux' 
import { addArticleAction, getData, loggedIn } from '../actions/addArticleAction'
import { OnlyTestSnap } from './OnlyTestSnap' 
// eslint-disable-next-line react/prefer-stateless-function
class LogIn extends React.PureComponent {
  
  constructor(props){
    super(props)
    this.submitForm = this.submitForm.bind(this)
    this.validateForm = this.validateForm.bind(this)
    this.usernameRef = React.createRef()
    this.passwordRef = React.createRef() 
  } 
  
  componentWillReceiveProps(nextProps){ 
       if(nextProps.isUserLoggedIn){   
              this.props.history.push('/home')
       } 
  }  

  validateForm(username, password){
    let formError = false
    let errors = {}
    if (username === ''){
      formError = true
      errors.username = 'Please Enter Your Username'
    }

    if (username.match(/^['a-zA-Z']+$/)){
      formError = true
      errors.username = 'Please Enter Username alphanumeric'
    }

    if (password === ''){
      formError = true
      errors.password = 'Please Enter Your Password'
    }
  
    return formError
  }

  submitForm(event) {
    event.preventDefault()
    let username = this.usernameRef.current.value
    let password = this.passwordRef.current.value
    let isFormError = this.validateForm(username, password)
    if (isFormError){
      alert('Something Went Wrong') 
      return false
    }

  
    this.props.loggedIn({ username: username, password: password })
    const { ...isUserLoggedIn } = this.props 
    console.log(' home this props', this.props); 
  
  }  

  render() {
  
    return (
      <>
        <Card>
          <OnlyTestSnap items = {[1]} />
          <CardBody>
            <CardTitle>Log In</CardTitle>
            <Form onSubmit={this.submitForm}>
              <FormGroup>
                <Label for='Username'>Username</Label>
                <span style={{ color: 'red' }}>
                  {' '}
              
                </span>
                <Input innerRef={this.usernameRef} type='text' placeholder='Please Enter Your Username' />
              </FormGroup> 
              <FormGroup>
                <Label for='Password'>Password</Label>
                <span style={{ color: 'red' }}>
                  {' '}
                </span>
                <Input innerRef={this.passwordRef} type='password' placeholder='Please Enter Your Password'/>
              </FormGroup>
              <Button>Submit</Button>
            </Form>
          </CardBody>
        </Card>
      </>
    )
  }
}
 
const mapStateToProps = (state) => {  
  return {isUserLoggedIn: state.fakeLoginData.isUserLoggedIn};
}        
 
const mapDispatchToProps = (dispatch) => {
  return {
    addArticleAction: (article) => {
      dispatch(addArticleAction(article))
    },
    getData: () => {
      dispatch(getData())
    },
    loggedIn: (userData) => {
      dispatch(loggedIn(userData))
    }
  }
}
 
export default connect(mapStateToProps, mapDispatchToProps)(LogIn)
