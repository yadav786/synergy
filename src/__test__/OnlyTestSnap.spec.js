import React from 'react'
import renderer from 'react-test-renderer'
import { OnlyTestSnap } from '../components/OnlyTestSnap'

it('renders correctly Only Test Snap ', () => {
  const items = [1, 2, 3, 4]
  const tree = renderer.create(<OnlyTestSnap items={items} />).toJSON()
  // console.log(tree)
  // console.log(tree.children[0].props)
  expect(tree).toMatchSnapshot()
})
