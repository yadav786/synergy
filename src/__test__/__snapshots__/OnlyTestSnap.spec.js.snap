// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Only Test Snap  1`] = `
<div>
  <p>
    Items Found 
    1
  </p>
  <p>
    Items Found 
    2
  </p>
  <p>
    Items Found 
    3
  </p>
  <p>
    Items Found 
    4
  </p>
</div>
`;
